[Huang Ching] iP

chingh20 opened the pull request:
# Duke project 
> If today is not the due day, today is not the do day 😊
> 
Welcome to Duke! 
Have you ever forgotten a deadline for homework or missed a test because you thought it was supposed to be on another day?
If `haveExperiencedAboved == true`, Duke is here to save you!


## The ultimate todo list 😄
Duke helps you keep track of your tasks and deadlines! Your tasks are put into a todo list and ultimately saved to a *file.txt* file on your device, so your list won't be gone even if you exit Duke! 
Here are several functions of Duke and commands you can use:
* See what is in your todo list:
   * **list**
* Add a task to your list:
   * **todo** **task description**
   * **event** *task description* **/at** *time description*
   * **deadline** *task description* **/by** *time description*
* Delete a task from your list:
   * **delete** *task number*
* Mark a task as done:
   * **done** *task number*
* Close Duke:
   * **bye**
   
## Setting up Duke
All you need to do is:
1. download it from [here](https://github.com/chingh20/ip).
2. start adding your tasks!

## About the tech
Duke is a project done with 
- [X] Java
- [X] JavaFX 
   

If you are a Java programmer, you can use it to practice Java! Here's the main method:
```javascript
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

chingh20 committed:
add echo function

chingh20 committed:
modify main

chingh20 committed:
add List function

chingh20 committed:
add done function

chingh20 committed:
create different kind of tasks

chingh20 committed:
include error messages

chingh20 committed:
add delete function

chingh20 committed:
add test cases

chingh20 committed:
add read file and create file function

chingh20 committed:
set time as localDate

chingh20 committed:
add writeToFile

chingh20 committed:
Merge branch 'branch-Level-7'

chingh20 committed:
fix merge conflict

chingh20 committed:
change deadline

chingh20 committed:
Merge branch 'branch-Level-8'

chingh20 committed:
extract classes

chingh20 committed:
put java files into package

chingh20 committed:
extract classes

chingh20 committed:
modify duke

chingh20 committed:
add test class

chingh20 committed:
add junit tests

chingh20 committed:
add comments and javadoc

chingh20 committed:
add method headers

chingh20 committed:
reformat code

chingh20 committed:
Reformat code

Reformatting code is important for the quality of the code.

chingh20 committed:
Add find function

Add parseFindCommand in Paser, findTask in TaskList, iterateMatchingTaskList in Ui.
Change runCommand in Command.

chingh20 committed:
Change method name for parsing find commands

chingh20 committed:
Change method names

chingh20 committed:
Merge branch 'A-JavaDoc'

chingh20 committed:
Resolve merge conflict

chingh20 committed:
Merge branch 'Level-9'

chingh20 committed:
Merge remote-tracking branch 'origin/add-gradle-support' into A-Gradle

chingh20 committed:
Add gradle to project

chingh20 committed:
Merge branch 'A-Gradle'

chingh20 committed:
Add javaFX and add getReplyFromDuke method in Command.java

Modify the original runNextCommand method so that it returns String instead of void.
Take out the side effects (print lines) from methods so each method is only responsible for one thing.

chingh20 committed:
Merge branch 'javaFX-tutorial' into branch-Level-10

chingh20 committed:
Merge branch 'branch-Level-10'

chingh20 committed:
Update README.md

marcusc55 commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,120 @@
+    Parser parser;
+    Storage storage;
+    TaskList taskList;
+    boolean canUseFilePath;
```
Should these variables be set to private instead?

marcusc55 commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,61 @@
+        db.flip();
+        return db;
+    }
+}
```
Should this file end with a new line?

marcusc55 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,55 @@
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
```
Should these unused commits be removed?

marcusc55 commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,12 @@
+/**
+ * Represents an error that only occurs for a Duke object.
+ */
+public class DukeException extends Exception{
```
Should the { be surrounded by whitespace?

marcusc55 commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,93 @@
+     */
+    @Override
+    public String saveTaskToFile() {
+        return this.getTypeOfTask() + "||" + this.getStatusIcon() + "||" + this.getDescription() + "||" + this.getTime();
```
Should this line be shortened to less than 120 characters?

marcusc55 commented on src/main/java/duke/Main.java:
```diff
@@ -0,0 +1,33 @@
+import javafx.scene.layout.AnchorPane;
+import javafx.stage.Stage;
+
+import java.io.IOException;
```
Should this import follow the import order of the coding standard?

marcusc55 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,135 @@
+            return "addTask";
+        } else if (str.startsWith("find")) {
+            return "find";
+        } else if (str.equals("\n")){
```
Should this { be surrounded with whitespace? I noticed the same in various other places too.

marcusc55 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,60 @@
+    protected String description;
+    protected boolean isDone;
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    public Task (String description, boolean isDone) {
+        this.description = description;
+        this.isDone = isDone;
+    }
```
Should these 2 constructors have JavaDoc comments?

marcusc55 reviewed (commented):
Some minor coding standard issues, but other than that your code looks good!

chingh20 committed:
Add assertions to the code

Add assetions to code where assumptions are made.

Adding assertions allow developers to detect errors in code when
the code is run. This improves the code quality.

chingh20 committed:
Reformat code

Extract methods and replace magic literals with variables.

Code quality is important, and by extracting methods and replacing
magic literals, the quality of code can be improved.

chingh20 committed:
Merge pull request #1 from chingh20/branch-A-CodeQuality

Reformat code

chingh20 committed:
Fix merge conflics

chingh20 committed:
Merge pull request #2 from chingh20/branch-A-Assertions

Add assertions

chingh20 committed:
Add extension to search list without exact matching keywords

chingh20 committed:
Edit date in deadline command

chingh20 committed:
Add comments to give credit to  * @author chingh20--reused

chingh20 committed:
Update README.md

chingh20 committed:
Update README

chingh20 committed:
Update README.md

chingh20 committed:
Extract methods to improve code quality

chingh20 committed:
Merge branch 'master' of http://github.com/chingh20/ip

chingh20 committed:
Add new images for user reference

chingh20 committed:
Update README.md

chingh20 committed:
Set theme jekyll-theme-leap-day

chingh20 committed:
Set theme jekyll-theme-cayman

chingh20 committed:
Set theme jekyll-theme-cayman

chingh20 committed:
Change application mainClassName in build.gradle

chingh20 committed:
Merge branch 'master' of http://github.com/chingh20/ip

chingh20 committed:
Change save list command

chingh20 committed:
Update README.md

chingh20 committed:
Change Ui demonstration picture

chingh20 committed:
Merge branch 'master' of http://github.com/chingh20/ip

chingh20 committed:
Update README.md

chingh20 committed:
Update README.md

chingh20 committed:
Update README.md

chingh20 committed:
Change Ui.png

chingh20 committed:
Merge branch 'master' of http://github.com/chingh20/ip

chingh20 committed:
Change Ui image

chingh20 committed:
Delete Ui.png

chingh20 committed:
Change Ui image

chingh20 committed:
Update README.md
